tune_sklearn/tune_gridsearch.py

```diff
@@ -192,6 +200,10 @@ def _tune_run(self, config, resources_per_trial):
                 `tune.run`.
 
         """
+        trainable = _Trainable
+        if check_is_pipeline(self.estimator) and self.early_stopping:
```

Comment:
if you plan on keeping `pipeline_detection => fit_last_stage`, don't forget to reference it here

```
if check... and self.early_stopping and self.fit_last_stage:
```
^ also for `TuneSearchCV`

Reply:
Hmm, `fit_last_stage` doesn't quite sound right either. Maybe just `pipeline_early_stopping`?

Reply:
`early_stop_last_stage` ? pipeline_early_stopping=False sounds like no early stopping will happen for Pipelines, but in fact the point of enabling it is to allow early_stopping to happen for `Pipeline` objects with partial fit right?

Reply:
It's actually the other way. It's supposed to handle pipelines *without* `partial_fit` (like the default sklearn one which is used by 99% of people). `coerce_pipeline`? convert doesn't seem right as there isn't any conversion involved.

Reply:
Got it, let me try to recap:

This is currently what we have today:
```
pipeline_detection (bool): Only relevant if estimator is Pipeline object and early_stopping is enabled/True. If True, early stopping will only be performed taking the last step of the pipeline into account. If False, early stopping will be determined by 'Pipeline.warm_start' or 'Pipeline.partial_fit' capabilities, which is not supported by standard SKlearn.
```

what I'm proposing is:
```
pipeline_auto_early_stop [or variant] (bool): Only relevant if estimator is Pipeline object and early_stopping is enabled/True. If True, early stopping will be performed on the last stage of the pipeline (which must support early stopping). If False, early stopping will be determined by 'Pipeline.warm_start' or 'Pipeline.partial_fit' capabilities, which is by default not supported by standard SKlearn.
```

`pipeline_auto_early_stop` actually works for me - let me know if that sounds good to you!